pyriemann/frequencybandselection.py

```diff
@@ -0,0 +1,173 @@
+"""Code for frequency band selection."""
+import numpy as np
+from mne import Epochs, events_from_annotations
+
+from .estimation import Covariances
+from .classification import class_distinctiveness
+
+
+def freq_selection_class_dis(raw, cv, freq_band=[5., 35.], sub_band_width=4,
+                             sub_band_step=2, tmin=0.5, tmax=2.5, picks=None,
+                             event_id=None,
+                             return_class_dis=False):
+    r"""Select optimal frequency band based on class distinctiveness measure.
+
+        Optimal frequency band is selected by combining a filter bank with
+        a heuristic based on class distinctiveness on the manifold:
+        1. Filter training raw EEG data for each sub-band of a filter bank.
+        2. Estimate covariance matrices of filtered EEG for each sub-band.
+        3. Measure the class distinctiveness of each sub-band using the
+        classDis metric.
+        4. Find the optimal frequency band by starting from the sub-band
+        with the largest classDis and expanding the selected frequency band
+        as long as the classDis exceeds the threshold :math:`th`:
+
+        .. math::
+        th=(max(classDis)−min(classDis))×0.4
+
+
+        This algorithm is described in [1]_.
+
+
+        Parameters
+        ----------
+        raw : Raw object
```

Comment:
I agree, to avoid introducing a new dependency in the core package, it is a good idea to put his file `examples/motor-imagery/helpers/` and to call it in the example. See for example [this one](https://github.com/pyRiemann/pyRiemann/blob/master/examples/motor-imagery/plot_ensemble_coherence.py#L38).

Reply:
I moved the function to the helpers. Please confirm it.